vio/viosslfactories.c

```diff
@@ -269,25 +274,40 @@ new_VioSSLFd(const char *key_file, const char *cert_file,
     goto err2;
   }
 
-  /* DH stuff */
+#if !defined(WOLFSSL)
+  /* DH stuff:
+
+     WolfSSL chooses right DH public key automatically depending
+     on RSA key size of server certificate during TLS handshake.
+     For OpenSSL >=  1.0.2 this feature needs to be explicitly turned on.
+     OpenSSL versions < 1.0.2 will still use the old get_dh2048() method
+     to obtain and set DH public key (see also MDEV-26015).
+  */
   if (!is_client_method)
   {
-    dh=get_dh2048();
+#if !defined(SSL_CTRL_SET_DH_AUTO)
```

Comment:
Same question [as above](https://github.com/MariaDB/server/pull/1869/files#r721693775) regarding whether or not an explicit check for OpenSSL >=1.0.2 is needed.